app/controllers/projects_complete_controller.rb

```diff
@@ -3,21 +3,16 @@ def complete
     @project = Project.find(project_id)
     authorize @project, :update?
 
-    if project_completable?
+    if @project.completable?
       set_project_completed_at
 
       render :completed
     else
-      flash[:alert] = I18n.t("project.complete.unable_to_complete_html")
+      flash[:alert] = I18n.t("#{@project.type.downcase.gsub("::", "_")}.complete.unable_to_complete_html")
```

Comment:
`@project.type.parameterize(separator: "_")` might give you what you want here?

Reply:
Ohhhh good tip